src/go/rpk/pkg/config/config.go

```diff
@@ -148,12 +147,14 @@ func AvailableModes() []string {
 func (c *Config) FileOrDefaults() *Config {
 	if c.File() != nil {
 		cfg := c.File()
-		cfg.loadedPath = c.loadedPath
-		cfg.ConfigFile = c.ConfigFile // preserve loaded ConfigFile property.
+		cfg.fileLocation = c.fileLocation
 		return cfg
 	} else {
 		cfg := Default()
-		cfg.ConfigFile = c.ConfigFile
+		// --config set but the file doesn't exist yet:
+		if c.fileLocation != "" {
+			cfg.fileLocation = c.fileLocation
+		}
```

Comment:
I thought the plan was to move this into `Load` once getting rid of ConfigFile?

Reply:
Changed, thanks :+1: